common/src/main/java/com/viaversion/viaversion/protocols/v1_21_9to1_21_11/rewriter/BlockItemPacketRewriter1_21_11.java

```diff
@@ -87,7 +90,11 @@ protected void handleItemDataComponentsToServer(final UserConnection connection,
         super.handleItemDataComponentsToServer(connection, item, container);
     }
 
-    public static void upgradeData(final Item item, final StructuredDataContainer container) {
+    public static void upgradeData(final ProtocolVersion serverVer, final Item item, final StructuredDataContainer container) {
```

Comment:
It shouldn't be in upgradeData but in an own function "appendItemDataFixComponents" like in other protocols (which is then called in handleItemDataComponentsToClient after upgradeData)